src/lib/ImageUpload/ImageUpload.tsx

```diff
@@ -0,0 +1,245 @@
+     * 
+     * @param e - Drag event information.
+     */
+    const onDrop = (e: DragEvent<HTMLInputElement>) => {
+        e.preventDefault();
+        setDragActive(false);
+        setError(undefined);
+        const acceptedFiles: File[] = [];
+        const declinedFiles: File[] = [];
+        const acceptedPreviews: string[] = [];
+        const dt = new DataTransfer();
+
+        if (e.dataTransfer.files) {
+            for (const file of Object.values(e.dataTransfer.files)) {
+                    const valid = isValidFile(file, acceptedFiles, declinedFiles, acceptedPreviews);
+                    if (valid) {
+                        dt.items.add(file);
+                    }
+            }
+        }
+
+        if (onImageSelected) {
+            onImageSelected(acceptedFiles[0]);
+        }
+        setUploadedImages(acceptedFiles);
+        setPreviews(acceptedPreviews);
+        if (inputRef.current) {
+            inputRef.current.files = dt.files.length ? dt.files : null;
+        }
+
+        if (declinedFiles.length > 0) {
+            setError(`${declinedFiles.length > 1
+                ? `${declinedFiles.length} files`
+                : 'File'} could not be loaded. Check for the correct file type and size (max size: ${maxSize}MB)`);
+        }
+    }
+
+    /**
+     * onDragOver event handler in order to support drag & drop functionality.
+     * Notifies when the dragged item has been dragged over the target view.
+     * 
+     * @param e - Drag event information
+     */
+    const onDragOver = (e: DragEvent<HTMLInputElement>) => {
+        e.preventDefault();
+        
+        if (!dragActive) {
+            setDragActive(true);
+        }
+    }
+
+    /**
+     * onDragLeave event handler in order to support drag & drop functionality.
+     * Notifies when the dragged item has left the target view.
+     * 
+     * @param e - Drag event information
+     */
+    const onDragLeave = (e: DragEvent<HTMLInputElement>) => {
+        if (dragActive) {
+            setDragActive(false);
+        }
+    }
+
+    useEffect(() => {
+        if (hasUploaded) {
+            clearInput();
+        }
+    }, [hasUploaded])
+
+    return <div className="c4imgupload--wrapper">
+        {previews && previews.length > 0 && <div className="c4imgupload--preview">
+            <div className="preview">
+                <img
+                    className="img"
+                    alt="Uploaded file preview"
+                    src={previews[0]}
+                />
+                <div className="cutout">
+                    <div className="left" />
+                    <div className="center" />
+                    <div className="right" />
+                </div>
+            </div>
+            {uploadedImages && uploadedImages.length > 0 && <div className="file-name">
+                <p>{uploadedImages[0].name}</p>
+            </div>}
+        </div>}
```

Comment:
Could we add a 'remove/cancel' red X icon for when there is an upload received but the user changes their mind? Again, not a blocker but definitely a nice to have when we have the time!

Reply:
For sure! Should be super quick!

Reply:
That is exactly what i meant! Thankyou!